centos_ci.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+# Copyright (C) 2016 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Output command before executing
+set -x
+
+# Exit on error
+set -e
+
+# Source environment variables of the jenkins slave
+# that might interest this worker.
+if [ -e "jenkins-env" ]; then
+  cat jenkins-env \
+    | grep -E "(JENKINS_URL|GIT_BRANCH|GIT_COMMIT|BUILD_NUMBER|ghprbSourceBranch|ghprbActualCommit|BUILD_URL|ghprbPullId)=" \
+    | sed 's/^/export /g' \
+    > ~/.jenkins-env
+  source ~/.jenkins-env
+fi
+
+# We need to disable selinux for now, XXX
+/usr/sbin/setenforce 0
+
+# Get all the deps in
+yum -y install \
+  gcc \
+  make \
+  git \
+  curl
+
+# Install Go 1.7
+curl -LO https://storage.googleapis.com/golang/go1.7.linux-amd64.tar.gz
+tar -xf go1.7.linux-amd64.tar.gz
+mv go /usr/local
+
+# Setup GOROOT
+export GOROOT=/usr/local/go
+# Setup GOPATH
+mkdir $HOME/gopath $HOME/gopath/src $HOME/gopath/bin $HOME/gopath/pkg
+export GOPATH=$HOME/gopath
+export PATH=$GOROOT/bin:$GOPATH/bin:$PATH
+
+# Setup minishift repo
+mkdir -p $GOPATH/src/github.com/minishift
+ln -s `pwd` $GOPATH/src/github.com/minishift/minishift
+cd $GOPATH/src/github.com/minishift/minishift
+
+# Setup Glide ("curl https://glide.sh/get | sh" not working due to https://github.com/Masterminds/glide/issues/708)
+GLIDE_OS_ARCH=`go env GOHOSTOS`-`go env GOHOSTARCH`
+GLIDE_TAG=$(curl -s https://glide.sh/version)
+GLIDE_LATEST_RELEASE_URL="https://github.com/Masterminds/glide/releases/download/${GLIDE_TAG}/glide-${GLIDE_TAG}-${GLIDE_OS_ARCH}.tar.gz"
+curl -LO ${GLIDE_LATEST_RELEASE_URL}
+mkdir /tmp/glide
+tar --directory=/tmp/glide -xvf glide-${GLIDE_TAG}-${GLIDE_OS_ARCH}.tar.gz
+export PATH=$PATH:/tmp/glide/${GLIDE_OS_ARCH}
+
+# Test
+make test cross fmtcheck prerelease
```

Comment:
Looking good. I was wondering whether you could look into one more thing though :-) The value proposition of using CentOS CI is to be able to run integration tests (under the assumption that we have nested virtualization on CentOS CI). To confirm this we could run the current integration tests as well.

To make this work you would need to add the _integration_ task here. But you also need to either provision VirtualBox (something I would like to investigate as well) or probably easier make a change to _flags.go_ to read the driver from say an environment variable and if not set defaults to _virtualbox_.